build.gradle

```diff
@@ -56,25 +56,61 @@ dependencies {
     testImplementation 'junit:junit:4.12'
 }
 
+// custom tasks
 task release(type: Zip) {
     dependsOn build
-    archiveFileName = "${project.name}-${readModInfoJson.version}.zip"
+    archiveFileName = "${project.name}-${readModInfoJson().version}.zip"
     destinationDirectory = file(rootDir)
     from sourceSets.main.resources
     from jar
     into project.name
 }
 
+task checkProperties {
+    doLast {
+        if (!areGradlePropertiesDefined()) {
+            throw new GradleException('Missing properties in gradle.properties. See gradle.properties.dist as an example.')
+        }
+    }
+}
+
 task copyToMods(type: Copy) {
-    if (!arePropertiesDefined) return
+    dependsOn checkProperties
     dependsOn build
     from jar
-    into getStarsectorModDir
+    into getStarsectorModDir()
 }
 
 task runGameWithBuild(type: Exec) {
-    if (!arePropertiesDefined) return
     dependsOn copyToMods
-    workingDir = getStarsectorExecutableDir
-    Os.isFamily(Os.FAMILY_WINDOWS) ? commandLine('cmd', '/c', 'starsector.bat') : commandLine('sh', './starsector.sh')
+    isWindows() ? commandLine('cmd', '/c', 'starsector.bat') : commandLine('sh', './starsector.sh')
+    doFirst {
+        workingDir = getStarsectorExecutableDir()
+    }
+}
+
+// methods
+def areGradlePropertiesDefined() {
+    return (project.hasProperty('game.dir') && project.hasProperty('game.mod_folder'))
+}
+
+def getStarsectorModDir() {
+    if (!areGradlePropertiesDefined()) {
```

Comment:
This is specifically for Copy task which needs to be set in configuration phase. Task still depends on checkProperties task so will throw correct exception if needed.